dali/python/nvidia/dali/pipeline.py

```diff
@@ -470,11 +476,20 @@ def feed_input(self, data_node, data, layout=""):
         if isinstance(data, list):
             inputs = []
             for datum in data:
-                inputs.append(Tensors.TensorCPU(datum, layout))
-            self._pipe.SetExternalTensorInput(name, inputs)
+                if hasattr(datum, "__cuda_array_interface__"):
+                    inp = Tensors.TensorGPU(datum, layout)
```

Comment:
I wondered why we don't specify the `device_id` argument here in any fashion, following up on that it appears to be set to current device in the constructor, as by default cupy is working on the current device (where all allocations and memory transfers happen). We discussed that it may be worth some comments.

Reply:
Done